ee/clickhouse/views/events.py

```diff
@@ -108,11 +111,14 @@ def list(self, request: Request, *args: Any, **kwargs: Any) -> Response:
 
         return Response({"next": next_url, "results": result})
 
-    def retrieve(self, request: Request, pk: Optional[int] = None, *args: Any, **kwargs: Any) -> Response:
-        query_result = sync_execute(SELECT_ONE_EVENT_SQL, {"team_id": self.team.pk, "event_id": pk},)
-        result = ClickhouseEventSerializer(query_result[0], many=False).data
-
-        return Response(result)
+    def retrieve(self, request: Request, pk: Optional[Union[int, str]] = None, *args: Any, **kwargs: Any) -> Response:
+        if not isinstance(pk, str) or not UUIDT.is_valid_uuid(pk):
+            return Response({"detail": "Invalid UUID", "code": "invalid", "type": "validation_error",}, status=400)
+        query_result = sync_execute(SELECT_ONE_EVENT_SQL, {"team_id": self.team.pk, "event_id": pk.replace("-", "")})
+        res: Union[Dict, str] = f"No events exist for event UUID {pk}"
```

Comment:
Sorry for keeping hammering on this PR - this makes for a bad API though.

In a previous comment I suggested `raise NotFound()`